Lib/test/test_ctypes/test_aligned_structures.py

```diff
@@ -0,0 +1,198 @@
+from ctypes import (
+    c_char, c_uint32, c_ubyte, alignment, sizeof,
+    BigEndianStructure, LittleEndianStructure,
+    BigEndianUnion, LittleEndianUnion,
+)
+import inspect
+import unittest
+
+
+class TestAlignedStructures(unittest.TestCase):
+    def test_aligned_string(self):
+        for base, data in (
+            (LittleEndianStructure, bytearray(
+                b'\x07\x00\x00\x00\x00\x00\x00\x00'
+                b'\x00\x00\x00\x00\x00\x00\x00\x00'
+                b'\x68\x65\x6c\x6c\x6f\x20\x77\x6f'
+                b'\x72\x6c\x64\x21\x00\x00\x00\x00'
+            )),
+            (BigEndianStructure, bytearray(
+                b'\x00\x00\x00\x07\x00\x00\x00\x00'
+                b'\x00\x00\x00\x00\x00\x00\x00\x00'
+                b'\x68\x65\x6c\x6c\x6f\x20\x77\x6f'
+                b'\x72\x6c\x64\x21\x00\x00\x00\x00'
+            )),
+        ):
+            class Aligned(base):
+                _align_ = 16
+                _fields_ = [
+                    ('value', c_char * 16),
+                ]
+
+            class Main(base):
+                _fields_ = [
+                    ('first', c_uint32),
+                    ('string', Aligned),
+                ]
+
+            main = Main.from_buffer(data)
+            self.assertEqual(main.first, 7)
+            self.assertEqual(main.string.value, b'hello world!')
+            self.assertEqual(
+                bytes(main.string.__buffer__(inspect.BufferFlags.SIMPLE)),
+                b'\x68\x65\x6c\x6c\x6f\x20\x77\x6f\x72\x6c\x64\x21\x00\x00\x00\x00'
+            )
+            self.assertEqual(Main.string.offset, 16)
+
+    def test_aligned_structures(self):
```

Comment:
Ah, the way that I have this structured, no it does not. I should have put this after say, another `c_ubyte` so that the structure is aligned at 4 bytes after a single byte rather than the 1 byte it should otherwise be aligned at.